dashboard/src/lib/porter-apps/index.ts

```diff
@@ -116,27 +116,40 @@ export function clientAppToProto(app: ClientPorterApp): PorterApp {
 
   const predeploy = app.services.find((s) => isPredeployService(s));
 
-  const proto = new PorterApp({
-    name: app.name,
-    services,
-    env: app.env,
-    build: {
-      context: app.build.context,
-      method: app.build.method,
-      buildpacks: app.build.buildpacks.map((b) => b.buildpack),
-      builder: app.build.builder,
-      dockerfile: app.build.dockerfile,
-    },
-    ...(app.image && {
-      image: {
-        repository: app.image.repository,
-        tag: app.image.tag,
-      },
-    }),
-    ...(predeploy && {
-      predeploy: serviceProto(serializeService(predeploy)),
-    }),
-  });
+  const proto = match(source)
+    .with(
+      { type: "github" },
+      () =>
+        new PorterApp({
+          name: app.name,
+          services,
+          env: app.env,
+          build: {
+            context: app.build.context,
+            method: app.build.method,
+            buildpacks: app.build.buildpacks.map((b) => b.buildpack),
+            builder: app.build.builder,
+            dockerfile: app.build.dockerfile,
+          },
+          ...(predeploy && {
+            predeploy: serviceProto(serializeService(predeploy)),
+          }),
```

Comment:
is this null safe? does having an empty/undefined predeploy key break things? 

personally I prefer a final object with determinate keys with default/empty values rather than one with nondeterminate keys 🤷 
```suggestion
            predeploy: serviceProto(serializeService(predeploy ?? {})),
```

Reply:
In this case, undefined is going to be allowed regardless since the generated types from protobuf assume all values are optional.

We also assume undefined/nil is meaningful on the CCP side of things. If it's nil, then we don't include it. With that being said, is there a default that would work in your mind?